frameworks/hdfs/tests/test_shakedown.py

```diff
@@ -24,36 +26,68 @@
 
 
 def setup_module(module):
-    install.uninstall(PACKAGE_NAME)
-    sdk_utils.gc_frameworks()
-    install.install(PACKAGE_NAME, DEFAULT_TASK_COUNT)
+    install.uninstall(FOLDERED_SERVICE_NAME, package_name=PACKAGE_NAME)
+    utils.gc_frameworks()
+    install.install(
+        PACKAGE_NAME,
+        DEFAULT_TASK_COUNT,
+        service_name=FOLDERED_SERVICE_NAME,
+        additional_options={"service": { "name": FOLDERED_SERVICE_NAME } })
+    plan.wait_for_completed_deployment(FOLDERED_SERVICE_NAME)
 
 
 def setup_function(function):
     check_healthy()
 
 
 def teardown_module(module):
-    install.uninstall(PACKAGE_NAME)
+    install.uninstall(FOLDERED_SERVICE_NAME, package_name=PACKAGE_NAME)
+
+@pytest.mark.sanity
+def test_endpoints():
+    # check that we can reach the scheduler via admin router, and that returned endpoints are sanitized:
+    core_site = etree.fromstring(cmd.run_cli('hdfs --name={} endpoints core-site.xml'.format(FOLDERED_SERVICE_NAME)))
+    check_properties(core_site, {
+        'ha.zookeeper.parent-znode': '/dcos-service-test__integration__hdfs/hadoop-ha'
+    })
+
+    hdfs_site = etree.fromstring(cmd.run_cli('hdfs --name={} endpoints hdfs-site.xml'.format(FOLDERED_SERVICE_NAME)))
+    expect = {
+        'dfs.namenode.shared.edits.dir': 'qjournal://' + ';'.join(['journal-{}-node.{}:8485'.format(i, FOLDERED_SERVICE_AUTOIP_HOST) for i in range(3)]) + '/hdfs',
+    }
+    for i in range(2):
+        expect['dfs.namenode.rpc-address.hdfs.name-{}-node'.format(i)] = 'name-{}-node.{}:9001'.format(i, FOLDERED_SERVICE_AUTOIP_HOST)
+        expect['dfs.namenode.http-address.hdfs.name-{}-node'.format(i)] = 'name-{}-node.{}:9002'.format(i, FOLDERED_SERVICE_AUTOIP_HOST)
+    check_properties(hdfs_site, expect)
+
+
+def check_properties(xml, expect):
+    found = {}
+    for prop in xml.findall('property'):
+        name = prop.find('name').text
+        if name in expect:
+            found[name] = prop.find('value').text
+    utils.out('expect: {}\nfound:  {}'.format(expect, found))
+    assert expect == found
 
 
 @pytest.mark.skip(reason="HDFS-451")
 @pytest.mark.data_integrity
 @pytest.mark.sanity
 def test_integrity_on_data_node_failure():
     shakedown.wait_for(
-        lambda: write_data_to_hdfs("data-0-node.hdfs.autoip.dcos.thisdcos.directory", TEST_FILE_1_NAME),
+        lambda: write_data_to_hdfs('data-0-node.{}'.format(FOLDERED_SERVICE_AUTOIP_HOST), TEST_FILE_1_NAME),
```

Comment:
Looking into this, because I'm doing a lot of the same things in the overlay tests. this is an example of when we want to use the `.mesos` DNS because we want the host IP to run the command on. Otherwise, when you call `shakedown.run_command_on_agent(<container_ip>...)` it won't work.